### ISSUE TITLE
Tutorial-style documentation for mini-batching via iXrange

### ISSUE THREAD

**Issue by stuhlmueller** [pyro-ppl/pyro#67]
This probably falls under general documentation (#12), but I wanted to create a separate issue since I expect that this will be one of the biggest sticking points for people coming from other probabilistic programming languages. (It certainly is for me.)

Something I'd find particularly helpful: Examples that show how to transform models that operate on individual data points into models that operate on batches. As an important special case, this should include sequence models that operate on inputs of varying lengths, such that we need to use padding/sorting.

Related: Support PyTorch DataLoader #46
